### DISCUSSION TITLE
"Returning the list of remaining integrals" with TARCER

### DISCUSSION THREAD

**Discussion by CarolinaPerdomo** [FeynCalc/feyncalc#110] [Q&A] [answered]
Hi, good evening,

I am computing one of the two-loop diagrams for the quark self-energy of QCD (in this case, the "QuarkBubble").

![image](https://user-images.githubusercontent.com/82186901/114109835-ffaa4a80-98cd-11eb-8ce8-034ae5a654b2.png)

After generating the amplitude with FeynArts and converted it to FeynCalc (and doing some trace evaluation) I got:

![image](https://user-images.githubusercontent.com/82186901/114109952-4c8e2100-98ce-11eb-8d94-da04ea19d165.png)

Now, I want to calculate the amplitude with TARCER. So,I used ToTFI[...] and TarcerRecurse[exp]:

![image](https://user-images.githubusercontent.com/82186901/114110160-c3c3b500-98ce-11eb-98fe-967b39550339.png)

Nevertheless, when I use TarcerExpand[], I got always the same output: "Returning the list of remaining integrals"

![image](https://user-images.githubusercontent.com/82186901/114110239-f4a3ea00-98ce-11eb-8b37-8ed349199df9.png)

After reading the arXiv about TARCER, I think there is still something I did not understand. Why TARCER couldn't reduce those integrals?

If someone can help me with this, I am grateful.
C.

**Comment by vsht** [comment, marked as answer]
Hi,

> Why TARCER couldn't reduce those integrals?

because those are your master integrals, so they can't be reduced any further.
It all makes sense, the integrals are a bubble, a sunrise and a dotted sunrise.

```
ints = {FAD[{p1 - q, m}, {p1 - p2, m}, {p2, m}], 
  FAD[{p1 - q, m, 2}, {p1 - p2, m}, {p2, m}], FAD[p1, {p1 - q, m}]}

tints = TarcerRecurse@ToTFI[#, {p1, p2}, {q}] & /@ ints

TarcerToFC[#, {p1, p2}] & /@ tints
```

> Nevertheless, when I use TarcerExpand[], I got always the same output: "Returning the list of remaining integrals"

TarcerExpand is not really helpful here, since it contains only analytic results for a handful of integrals (cf. `$BasisIntegrals`)
Your integrals are off-shell, so here one anyhow needs to get the results from elsewhere (literature, other packages, ...)

Cheers,
Vladyslav

**Reply by CarolinaPerdomo** [reply]
Hi Vladyslav! thank you so much for answering my question, I am calmer now. Cheers!

**Comment by huansouza** [comment]
Hey @CarolinaPerdomo,

You can find the integrals you need in [0501132](https://arxiv.org/abs/hep-ph/0501132) and [0307101](https://arxiv.org/abs/hep-ph/0307101). If you are interested, I and some collaborators have been working with TARCER to compute such diagrams, you can take a look at our work here: [2105.12577](https://arxiv.org/abs/2105.12577). The Mathematica notebook is available at https://github.com/lehum/einstein-qed

Best wishes,

Huan.
